demos/smart_classroom_demo_gapi/main.cpp

```diff
@@ -1,1046 +1,328 @@
-
-    gflags::ParseCommandLineNonHelpFlags(&argc, &argv, true);
-    if (FLAGS_h) {
-        showUsage();
-        showAvailableDevices();
-        return false;
-    }
-
-    slog::info << "Parsing input parameters" << slog::endl;
-
-    if (FLAGS_i.empty()) {
-        throw std::logic_error("Parameter -i is not set");
-    }
-    if (FLAGS_m_act.empty() && FLAGS_m_fd.empty()) {
-        throw std::logic_error("At least one parameter -m_act or -m_fd must be set");
-    }
-
-    return true;
-}
-
-}  // namespace
-
-int main(int argc, char* argv[]) {
+int main(int argc, char* argv[]) { 
     try {
         /** This demo covers 4 certain topologies and cannot be generalized **/
-        slog::info << "InferenceEngine: " << printable(*GetInferenceEngineVersion()) << slog::endl;
-
-        if (!ParseAndCheckCommandLine(argc, argv)) {
+        if (!util::ParseAndCheckCommandLine(argc, argv)) {
             return 0;
         }
 
-        const auto ad_model_path = FLAGS_m_act;
-        const auto fd_model_path = FLAGS_m_fd;
-        const auto fr_model_path = FLAGS_m_reid;
-        const auto lm_model_path = FLAGS_m_lm;
-        const auto teacher_id = FLAGS_teacher_id;
+        /** Prepare parameters **/
+        const std::string video_path = FLAGS_i;
+        const auto ad_model_path     = FLAGS_m_act;
+        const auto fd_model_path     = FLAGS_m_fd;
+        const auto fr_model_path     = FLAGS_m_reid;
+        const auto lm_model_path     = FLAGS_m_lm;
 
-        if (!FLAGS_teacher_id.empty() && !FLAGS_top_id.empty()) {
-            slog::err << "Cannot run simultaneously teacher action and top-k students recognition."
-                      << slog::endl;
-            return 1;
-        }
+        /** Print info about demo's properties **/
+        config::printInfo();
 
-        const auto actions_type = FLAGS_teacher_id.empty()
-                                      ? FLAGS_a_top > 0 ? TOP_K : STUDENT
-                                      : TEACHER;
-        const auto actions_map = actions_type == STUDENT
-                                     ? split(FLAGS_student_ac, ',')
-                                     : actions_type == TOP_K
-                                         ? split(FLAGS_top_ac, ',')
-                                         : split(FLAGS_teacher_ac, ',');
-        const auto num_top_persons = actions_type == TOP_K ? FLAGS_a_top : -1;
-        const auto top_action_id = actions_type == TOP_K
-                                   ? std::distance(actions_map.begin(), find(actions_map.begin(), actions_map.end(), FLAGS_top_id))
-                                   : -1;
-        if (actions_type == TOP_K && (top_action_id < 0 || top_action_id >= static_cast<int>(actions_map.size()))) {
-            slog::err << "Cannot find target action: " << FLAGS_top_id << slog::endl;
-            return 1;
-        }
+        cv::VideoCapture cap(video_path != "cam" ? video_path : 0);
```

Comment:
not sure how this `?:` works but barely remember that both true- and false- branch should have the same expression type. MAYBE it results into a proper constructor overload being called but I don't really know.